src/proxy/mod.rs

```diff
@@ -74,24 +82,34 @@ impl Server {
 
 async fn proxy(
     req: Request<hyper::body::Incoming>,
-    backend_addr: SocketAddr,
+    server: Arc<Server>,
 ) -> Result<Response<BoxBody<Bytes, hyper::Error>>, hyper::Error> {
+    let backend_addr = server.get_next().to_string();
```

Comment:
This function is pretty hard to reach through. One solution is to break this function down into smaller functions. Is that possible?